### DISCUSSION TITLE
Vercel Analytics Audience API

### DISCUSSION THREAD

**Discussion by debater-coder** [vercel/community#1068] [Idea]
Hi,

I've just tried out your new audiences feature, and it works really well. I just wish there was a feature to access audience data programmatically through an API.

**Comment by amyegan** [comment]
Thanks for reaching out @debater-coder! I shared this feedback with the analytics team 🙂

**Comment by timolins** [comment]
Hi @debater-coder. Love that idea. Mind sharing what you would use the API for?

**Reply by debater-coder** [reply]
I have been thinking of using that data for a page view counter, and for exporting to collect further insights (average, standard deviation, etc).

**Comment by lord007tn** [comment]
We are selling b2b2c solutions and all of our businesses' websites are developed and managed by our startup. we would like to add audience analytics in our Saas back office so the businesses can access their data. and why not chained it with some marketing tools to improve decision-making.

**Reply by timolins** [reply]
Hi @lord007tn. Thanks for your suggestion. Just to clarify, you want to expose analytics data you collect on your customers behalf, to your customers? Would you an API work for your use-case, or would you rather have an UI?

**Reply by lord007tn** [reply]
just An API we already developed our own UI integrated in our saas, also API is useful to combine with other data from other solutions.

**Reply by timolins** [reply]
Got it, thanks for clarifying. We'll keep your feedback in mind when laying out our API. 👍

**Comment by shandysulen** [comment]
I'm interested in surfacing advertising slots in my website, so I'd like to transparently share with my users and potential customers the reach of my website. A programmatic Audience API would be awesome for this.

**Comment by mattwigham** [comment]
We're launching a site-builder platform on Vercel and would also love an API for audience analytics to bake into our customer dashboards, showing data specific to their domain.

**Reply by cameronking4** [reply]
SAME!! need api call to vercel analytics

**Comment by adisher** [comment]
I have a portfolio website build with next.js that uses appDir and deployed on Vercel. I have enabled audiences analytics and would like to display the views on my landing page for credibility towards monthly viewership.

**Comment by onwp** [comment]
I am building a SaaS product where I would love to display Audiences on customer dashboards. @amyegan

**Reply by onwp** [reply]
Please make it usable through an API.

**Comment by FabioFreitasP** [comment]
I would like to have this available as well. I will use it to display information from multiple platforms in the same dashboard.

**Comment by reubenjh** [comment]
+1 . Would love to be able to access analytics data through an API. I would add it to my company's admin dashboard for staff to track engagement and traffic over time, see its relationship to our marketing projects, see growth etc

**Comment by mjovanc** [comment]
Would love to have that feature as well. Need to have a view counter for a website.

**Comment by jonxuxu** [comment]
Same, an easy way to display page views using the Analytics API would be awesome!!

**Comment by reubenjh** [comment]
FYI i'm probably reverting my vercel analytics subscription as I need this API and can do it (although clunkily...) through google analytics. Doesnt exactly look like this is coming anytime soon

**Comment by djsjr** [comment]
Could we have a rough timeframe on this and (importantly) which plans this feature will be available for? Will be consequential to whether or not I'm able to go further with Vercel hosting!

**Comment by Rafath-Auvee** [comment]
What's the update on this analytics export feature? I would like to export my analytics report day to day basis on excel or pdf files.

**Comment by bastienrobert** [comment]
There's a way to get the data in JSON by accessing this URL while you're logged or using any HTTP client w/ a Vercel Token in the Authorization header: 
```
https://vercel.com/api/web/insights/stats/CATEGORY?environment=production&filter=%7B%7D&from=FROM_DATETIME&limit=250&projectId=PROJECT_ID&teamId=TEAM_ID&to=TO_DATETIME

CATEGORY: path, referrer, country, os_name, client_name
PROJECT_ID
TEAM_ID
FROM_DATETIME, TO_DATETIME (eg.): 2023-10-12T00:00:00.000+02:00
```

Don't forget to encode the date ([this tool is made for that](https://www.urlencoder.org/))

**Comment by mikerudge** [comment]
We just started using analytics, and it was a breeze to set up. Great work! 

However, we need to show other stakeholders in the business some of the stats now. We need an API or data export option that we can run through some BI software (Power BI in our case) 

Thanks!

**Comment by paka747** [comment]
Access to analytics from the API level would be a great solution, in the case of my project it is a key issue, so without such an option I cannot use your analytic tool.

**Comment by joel-shor** [comment]
I get the following error. Since this isn't an exposed API, perhaps the internals of the analytics changed so this hack is no longer valid?

"Invalid request: `type` should be equal to one of the allowed values \"path, referrer, referrer_hostname, country, client_name, os_name, event_name, event_property, event_property_value, event_data, query_params, hostname, route\"

**Reply by bastienrobert** [reply]
Please give some context, maybe the request URL you're sending?

**Comment by andreeasteriu** [comment]
any updates about this subject?

**Reply by bastienrobert** [reply]
https://github.com/vercel/community/discussions/1068#discussioncomment-7403107

**Reply by amyegan** [reply]
I don't have an ETA for a stable public API. The undocumented API endpoint shared above can be used as a workaround, but be aware that it could change at any time without notice.

**Comment by cameronking4** [comment]
Need help in 2024 on this! I want to retrieve my vercel analytics for a project via API so I can use in my own internal system

**Reply by bastienrobert** [reply]
https://github.com/vercel/community/discussions/1068#discussioncomment-7403107

**Reply by amyegan** [reply]
Just be aware that the API endpoint shared above is not stable and could change at any time without notice 🙏

**Comment by versecafe** [comment]
I would like to share a convenience project for this, https://vaas-brown.vercel.app/ VAAS: Vercel Analytics API Scraper, I spent far too much time on UI design for this but hey what's the fun in making an API UI if it's not pretty.

<img width="1379" alt="image" src="https://github.com/vercel/community/assets/147033096/32947fa8-d570-4518-b30e-a44951f621c9">
<img width="1379" alt="image" src="https://github.com/vercel/community/assets/147033096/7469864f-5f2b-422a-b67a-00dd61bd8dc7">

This is still under early development and could vanish if the API changes but for now its here and I'll be adding features day by day

@mikerudge not sure if it's up to what you need right now for data export but does allow easy export in JSON of view, visitor data over time and will be adding more in the coming days

**Comment by Jordaneisenburger** [comment]
I would also like an API to get the data programmatically, is this somewhere on a roadmap or a feature very far out into the future ?

**Reply by versecafe** [reply]
@Jordaneisenburger  there is an unofficial endpoint that the VAAS project uses and I have a simple TS library wrapper around it and I have a experimental python library for it as well (also have one for Speed Insights) if you'd be interested in either I can publish them to `npm` also @CBrian04 would you be interested in this as well?

**Reply by reubenjh** [reply]
I'd be interested to use the internal workings of that system in my admin interface until they build these proper (if ever). However, probably not going to plug an access token into a random website I don't control 😆 Thanks for the effort making it though! Love your work

**Reply by versecafe** [reply]
@reubenjh I'm working on moving it to a turborepo to actually publish out the internal logic as a npm package lol, I get not wanting to put a token in the project is open source too so you can see that we don't track it but very understandable.

**Reply by Jordaneisenburger** [reply]
HI @versecafe, I'd even love to see just a simple GH repo if possible?

**Reply by versecafe** [reply]
@Jordaneisenburger https://github.com/versecafe/vaas in the middle of a move into turborepo so the main app will move into `apps/vaas` soon

**Comment by CBrian04** [comment]
I agree totally with others, it is more a **necessity** than a functionnality as 90% of time the person who does have to check statistics about a website have not access to vercel account but instead to the admin interface where developpers should be able to display data collected as they want.
